### DISCUSSION TITLE
How to write tests for Magefiles?

### DISCUSSION THREAD

**Discussion by mloskot** [magefile/mage#492] [Q&A]
I'd like to learn how to test the Go code written into my Magefiles.

Do I [add a test](https://go.dev/doc/tutorial/add-a-test) as `<magefile>_test.go` and use `go test` as for any other Go application?
Are there any Mage-specific limitations or requirements?

**Comment by sheldonhull** [comment]
I have some examples. I primarily use the examples test feature since it's integration style code when working with mage. 

Maybe this will help you. 

https://github.com/sheldonhull/magetools

Take a peek at some of the tests. They are slow but work well to document and confirm basic testing functionality.

**Reply by mloskot** [reply]
Thanks @sheldonhull I'll have a look at your tools
